src/object/background.cpp

```diff
@@ -418,11 +415,18 @@ Background::draw(DrawingContext& context)
 
   if (!m_image)
     return;
+    
+  context.push_transform();
+  if (!context.perspective_scale(m_parallax_speed.x, m_parallax_speed.y)) {
+    //The background is placed behind the camera.
+    context.pop_transform();
+    return;
+  }
+  context.set_flip(context.get_flip() ^ m_flip);
 
   Sizef level_size(d_gameobject_manager->get_width(),
                    d_gameobject_manager->get_height());
-  Sizef screen(context.get_width(),
-               context.get_height());
+  Sizef screen = context.get_viewport().get_size();
```

Comment:
Can this be a reference?

Reply:
`DrawingContext::get_viewport` returns an instance of `Rect&`.
`Rect::get_size` returns an instance of `Size`, which is not a reference type.
`Rect::get_size` cannot be modified to return an instance of `Size&`, because the result is calculated on demand and is not stored within the class.

Reply:
`screen` can be `const` though.

Reply:
const ref, right?

Reply:
Hume explained why it cannot be a reference above.

Reply:
true, I apparently can't keep focus.

Reply:
I made it const, together with a few more variables.